WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -0,0 +1,67 @@
+import UIKit
+import Yosemite
+
+/// Controls navigation for the flow to add a product given a navigation controller.
+///
+final class AddProductCoordinator: Coordinator {
+    var navigationController: UINavigationController
+
+    private let siteID: Int64
+    private let sourceView: UIBarButtonItem
+
+    init(siteID: Int64, sourceView: UIBarButtonItem, sourceNavigationController: UINavigationController) {
+        self.siteID = siteID
+        self.sourceView = sourceView
+        self.navigationController = sourceNavigationController
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func start() {
+        presentProductTypeBottomSheet()
+    }
+}
+
+// MARK: Navigation
+private extension AddProductCoordinator {
+    func presentProductTypeBottomSheet() {
+        let title = NSLocalizedString("Select a product type",
+                                      comment: "Message title of bottom sheet for selecting a product type to create a product")
+        let viewProperties = BottomSheetListSelectorViewProperties(title: title)
+        let command = ProductTypeBottomSheetListSelectorCommand(selected: nil) { selectedProductType in
+            self.navigationController.dismiss(animated: true) {
+                // Strong reference to `self` is required since `AddProductCoordinator` is not strongly referenced by any class.
+                self.presentProductForm(productType: selectedProductType)
+            }
+        }
+        // Until we support adding a variation, adding a variable product is disabled.
+        command.data = [.simple, .grouped, .affiliate]
+        let productTypesListPresenter = BottomSheetListSelectorPresenter(viewProperties: viewProperties, command: command)
+        productTypesListPresenter.show(from: navigationController, sourceBarButtonItem: sourceView, arrowDirections: .up)
+    }
+
+    func presentProductForm(productType: ProductType) {
+        let product = ProductFactory().createNewProduct(type: productType, siteID: siteID)
+        let model = EditableProductModel(product: product)
+
+        let currencyCode = ServiceLocator.currencySettings.currencyCode
+        let currency = ServiceLocator.currencySettings.symbol(from: currencyCode)
+        let productImageActionHandler = ProductImageActionHandler(siteID: product.siteID,
+                                                                  product: model)
+        let viewModel = ProductFormViewModel(product: model,
+                                             formType: .add,
+                                             productImageActionHandler: productImageActionHandler,
+                                             isEditProductsRelease3Enabled: true)
```

Comment:
Is this true, because the usage of this coordinator is behind the feature flag?

Reply:
yes! adding products is M4 and we always make the previous milestones publicly available 🙂